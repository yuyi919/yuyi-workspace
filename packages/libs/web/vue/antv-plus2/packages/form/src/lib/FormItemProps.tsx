import { CSSProperties, Types } from "@yuyi919/shared-types";
import { Component, extractUnsafeProps, Prop } from "@antv-plus2/helper";
import { isNum } from "@antv-plus2/shared";
import { autoSizer } from "@antv-plus2/theme";
import { Icon } from "ant-design-vue";
import { LayoutItemProps } from "./Layout";

type Renderer = any;

@Component()
export class FormItemProps extends LayoutItemProps {
  @Prop(String)
  prefixCls?: string;
  /**
   * 展示标签
   */
  @Prop()
  label?: Renderer;
  /**
   * 问号提示
   */
  @Prop()
  tooltip?: Renderer;
  /**
   * 标签样式
   */
  @Prop(Object)
  labelStyle?: CSSProperties;
  /**
   * 组件容器样式
   */
  @Prop(Object)
  wrapperStyle?: CSSProperties;
  /**
   * 组件容器样式
   */
  @Prop()
  addonBefore?: Renderer;
  /**
   * 组件容器样式
   */
  @Prop()
  addonAfter?: Renderer;
  /**
   * 组件容器样式
   */
  @Prop()
  extra?: Renderer;
  /**
   * 组件容器样式
   */
  @Prop()
  feedbackText?: Renderer;
  /**
   * 组件容器样式
   */
  @Prop(String)
  feedbackStatus?: "error" | "warning" | "success" | "pending" | Types.DynamicString;
  /**
   * 组件容器样式
   */
  @Prop()
  feedbackIcon?: Renderer;
  /**
   * 组件容器样式
   */
  @Prop(Boolean)
  asterisk?: boolean;
}
export default FormItemProps;

export const [FormItemPropConfig, useFormLayoutItemProps] = extractUnsafeProps(
  FormItemProps,
  (props) => {
    const {
      feedbackStatus,
      labelCol,
      wrapperCol,
      labelStyle: { ...labelStyle } = {},
      labelWidth,
      wrapperWidth,
      wrapperStyle: { ...wrapperStyle } = {},
    } = props;
    const feedbackIcon =
      props.feedbackIcon ??
      (feedbackStatus === "pending" ? (
        <Icon type="loading" theme="outlined" style={{ color: "#1890ff" }} />
      ) : (
        void 0
      ));

    // 是否启用了栅格
    const enableCol = props.enableCol !== false ? isNum(labelCol) || isNum(wrapperCol) : false;
    if (labelWidth) {
      labelStyle.width = labelWidth === "auto" ? undefined : autoSizer(labelWidth);
      labelStyle.maxWidth = labelWidth === "auto" ? undefined : autoSizer(labelWidth);
    }
    if (wrapperWidth) {
      wrapperStyle.width = wrapperWidth === "auto" ? undefined : autoSizer(wrapperWidth);
      wrapperStyle.maxWidth = wrapperWidth === "auto" ? undefined : autoSizer(wrapperWidth);
    }
    return {
      ...props,
      feedbackIcon,
      wrapperStyle,
      labelStyle,
      enableCol,
    };
  }
);
