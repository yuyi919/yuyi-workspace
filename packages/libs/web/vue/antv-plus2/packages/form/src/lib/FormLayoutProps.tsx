import { LayoutProps } from "./Layout";
import { Component, Prop } from "@yuyi919/vue-antv-plus2-helper";

@Component()
export class FormLayoutProps extends LayoutProps {
  @Prop()
  prefixCls?: string;
}
