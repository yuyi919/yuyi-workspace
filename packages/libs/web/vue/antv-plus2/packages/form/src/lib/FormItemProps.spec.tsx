import { FormItemProps, useFormLayoutItemProps } from "./FormItemProps";
import { FormLayoutProps } from "./FormLayoutProps";
describe("FormItemProps", () => {
  it("收集混合Props", () => {
    const props: FormItemProps = { feedbackStatus: "pending" },
      layoutProps: FormLayoutProps = { layout: "horizontal" };
    const usedProps = useFormLayoutItemProps(props, (key, value, option) => {
      return key in layoutProps ? value ?? layoutProps[key as keyof typeof layoutProps] : value;
    });
    expect(usedProps).toMatchSnapshot();
  });
});
