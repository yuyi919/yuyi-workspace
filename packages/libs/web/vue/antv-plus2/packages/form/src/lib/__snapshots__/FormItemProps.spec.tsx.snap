// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItemProps 收集混合Props 1`] = `
Object {
  "addonAfter": undefined,
  "addonBefore": undefined,
  "asterisk": undefined,
  "bordered": true,
  "colon": true,
  "extra": undefined,
  "feedbackIcon": VNode {
    "asyncFactory": undefined,
    "asyncMeta": undefined,
    "children": undefined,
    "componentInstance": undefined,
    "componentOptions": Object {
      "Ctor": [Function],
      "children": Array [],
      "listeners": Object {},
      "propsData": Object {
        "theme": "outlined",
        "type": "loading",
      },
      "tag": undefined,
    },
    "context": Vue {
      "$attrs": Object {},
      "$children": Array [],
      "$createElement": [Function],
      "$listeners": Object {},
      "$options": Object {
        "_base": [Function],
        "beforeCreate": Array [
          [Function],
        ],
        "components": Object {},
        "directives": Object {},
        "filters": Object {},
        "mounted": Array [
          [Function],
        ],
        "updated": Array [
          [Function],
        ],
      },
      "$parent": undefined,
      "$refs": Object {},
      "$root": [Circular],
      "$scopedSlots": Object {},
      "$slots": Object {},
      "$vnode": undefined,
      "_c": [Function],
      "_data": Object {},
      "_directInactive": false,
      "_events": Object {},
      "_hasHookEvent": false,
      "_inactive": null,
      "_isBeingDestroyed": false,
      "_isDestroyed": false,
      "_isMounted": false,
      "_isVue": true,
      "_renderProxy": Vue {
        "$attrs": Object {},
        "$children": Array [],
        "$createElement": [Function],
        "$listeners": Object {},
        "$options": Object {
          "_base": [Function],
          "beforeCreate": Array [
            [Function],
          ],
          "components": Object {},
          "directives": Object {},
          "filters": Object {},
          "mounted": Array [
            [Function],
          ],
          "updated": Array [
            [Function],
          ],
        },
        "$parent": undefined,
        "$refs": Object {},
        "$root": [Circular],
        "$scopedSlots": Object {},
        "$slots": Object {},
        "$vnode": undefined,
        "_c": [Function],
        "_data": Object {},
        "_directInactive": false,
        "_events": Object {},
        "_hasHookEvent": false,
        "_inactive": null,
        "_isBeingDestroyed": false,
        "_isDestroyed": false,
        "_isMounted": false,
        "_isVue": true,
        "_renderProxy": [Circular],
        "_self": [Circular],
        "_staticTrees": null,
        "_uid": 0,
        "_vnode": null,
        "_watcher": null,
        "_watchers": Array [],
      },
      "_self": [Circular],
      "_staticTrees": null,
      "_uid": 0,
      "_vnode": null,
      "_watcher": null,
      "_watchers": Array [],
    },
    "data": Object {
      "attrs": undefined,
      "class": undefined,
      "domProps": undefined,
      "hook": Object {
        "destroy": [Function],
        "init": [Function],
        "insert": [Function],
        "prepatch": [Function],
      },
      "key": undefined,
      "model": undefined,
      "nativeOn": undefined,
      "on": undefined,
      "props": Object {
        "theme": "outlined",
        "type": "loading",
      },
      "ref": undefined,
      "scopedSlots": Object {},
      "slot": undefined,
      "style": Object {
        "color": "#1890ff",
      },
    },
    "elm": undefined,
    "fnContext": undefined,
    "fnOptions": undefined,
    "fnScopeId": undefined,
    "isAsyncPlaceholder": false,
    "isCloned": false,
    "isComment": false,
    "isOnce": false,
    "isRootInsert": true,
    "isStatic": false,
    "key": undefined,
    "ns": undefined,
    "parent": undefined,
    "raw": false,
    "tag": "vue-component-1-AIcon",
    "text": undefined,
  },
  "feedbackLayout": "loose",
  "feedbackStatus": "pending",
  "feedbackText": undefined,
  "fullness": undefined,
  "gridSpan": undefined,
  "inset": undefined,
  "label": undefined,
  "labelAlign": undefined,
  "labelCol": undefined,
  "labelStyle": undefined,
  "labelWidth": undefined,
  "labelWrap": undefined,
  "layout": "horizontal",
  "prefixCls": undefined,
  "size": "default",
  "tooltip": undefined,
  "tooltipLayout": "icon",
  "wrapperAlign": undefined,
  "wrapperCol": undefined,
  "wrapperStyle": undefined,
  "wrapperWidth": undefined,
  "wrapperWrap": undefined,
}
`;
